src/main/kotlin/no/nav/familie/ef/sak/vilkår/PatchVilkårController.kt

```diff
@@ -12,43 +13,61 @@ import org.springframework.web.bind.annotation.PostMapping
+                            private val fagsakService: FagsakService,
+                            private val fagsakPersonService: FagsakPersonService) {
 
     private val secureLogger = LoggerFactory.getLogger("secureLogger")
 
-    @PostMapping("dokumentasjon-tilsynsutgifter")
-    fun patchDokumentasjonTilsynsutgifter(@RequestBody liveRun: LiveRun)
+    @PostMapping("inntekt")
+    fun patchInntekt(@RequestBody liveRun: LiveRun)
             : Ressurs<String> {
 
         val behandlinger = patchVilkårRepository.finnBehandlingerSomHarBarnetilsyn()
         behandlinger.forEach {
             val eksisterendeVilkår = vilkårsvurderingRepository.findByBehandlingId(it.id)
-            if (eksisterendeVilkår.map { it.type }.contains(VilkårType.DOKUMENTASJON_TILSYNSUTGIFTER)) {
-                secureLogger.info("det finnes allerede et vilkår for dokumentasjon av tilsynsutgifter på behandling med id=${it.id}")
+
+            if (eksisterendeVilkår.flatMap { it.delvilkårsvurdering.delvilkårsvurderinger.flatMap { it.vurderinger.map { it.regelId } } }
+                            .contains(RegelId.INNTEKT_SAMSVARER_MED_OS)) {
+                secureLogger.info("det finnes allerede et delvilkår for om inntekt samsvarer med OS på behandling med id=${it.id}")
             } else {
-                secureLogger.info("oppretter vilkår for dokumentasjon av tilsynsutgifter på behandling med id=${it.id}")
-                val nyVilkårsvurdering = Vilkårsvurdering(behandlingId = it.id,
-                                                          resultat = Vilkårsresultat.OPPFYLT,
-                                                          type = VilkårType.DOKUMENTASJON_TILSYNSUTGIFTER,
-                                                          delvilkårsvurdering = DelvilkårsvurderingWrapper(delvilkårsvurderinger = listOf(
-                                                                  Delvilkårsvurdering(resultat = Vilkårsresultat.OPPFYLT,
-                                                                                      vurderinger = listOf(Vurdering(regelId = RegelId.HAR_DOKUMENTERTE_TILSYNSUTGIFTER,
-                                                                                                                     svar = SvarId.JA,
-                                                                                                                     begrunnelse = ""))))))
-                if (liveRun.skalPersistere) {
-                    secureLogger.info("persistererer vilkår for dokumentasjon av tilsynsutgifter på behandling med id=${it.id}")
-                    vilkårsvurderingRepository.insert(nyVilkårsvurdering)
+                secureLogger.info("bygger opp nytt delvilkår for om inntekt samsvarer med OS på behandling med id=${it.id}")
+                val gammelVilkårsvurderingEllerNull = eksisterendeVilkår.find { vilkår -> vilkår.type == VilkårType.INNTEKT }
```

Comment:
Her mener jeg alle burde ha delvilkår inntekt - så kanskje ` ?: error("Mangler vilkårtype Inntekt for behandling ${it.id}")` er greit å slenge på på slutten?